File: api/src/scrub/dto/jobs.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel


class JobProgressResponse(BaseModel):
    """Response model for job progress."""

    request_id: str
    status: str
    total_items: int
    processed_items: int
    valid_count: int
    invalid_count: int
    error_count: int
    progress_percentage: int
    is_complete: bool
    error_message: Optional[str] = None


class ActiveJobItem(BaseModel):
    """Item in active jobs list."""

    request_id: str
    job_type: str
    status: str
    list_name: Optional[str]
    total_items: int
    processed_items: int
    progress_percentage: int
    created_at: datetime

File: web/types/jobs.ts
export interface JobProgressResponse {
  request_id: string;
  status: string;
  total_items: number;
  processed_items: number;
  valid_count: number;
  invalid_count: number;
  error_count: number;
  progress_percentage: number;
  is_complete: boolean;
  error_message: string | null;
}

export interface ActiveValidationJob {
  requestId: string;
  jobType: string;
  status: string;
  listName: string | null;
  totalItems: number;
  processedItems: number;
  progressPercentage: number;
  createdAt: string;
}

export interface ActiveJobItem {
  request_id: string;
  job_type: string;
  status: string;
  list_name: string | null;
  total_items: number;
  processed_items: number;
  progress_percentage: number;
  created_at: string;
}
